testsys/src/run_vmware.rs

```diff
@@ -0,0 +1,255 @@
+
+/// Create an EKS resource, EC2 resource and run Sonobuoy.
+#[derive(Debug, StructOpt)]
+pub(crate) struct RunVmware {
+    /// Name of the sonobuoy test.
+    #[structopt(long, short)]
+    name: String,
+
+    /// Location of the sonobuoy test agent image.
+    // TODO - default to an ECR public repository image
+    #[structopt(long, short)]
+    test_agent_image: String,
+
+    /// Name of the pull secret for the sonobuoy test image (if needed).
+    #[structopt(long)]
+    test_agent_pull_secret: Option<String>,
+
+    /// Keep the test agent running after completion.
+    #[structopt(long)]
+    keep_running: bool,
+
+    /// The plugin used for the sonobuoy test. Normally this is `e2e` (the default).
+    #[structopt(long, default_value = "e2e")]
+    sonobuoy_plugin: String,
+
+    /// The mode used for the sonobuoy test. One of `non-disruptive-conformance`,
+    /// `certified-conformance`, `quick`. Although the Sonobuoy binary defaults to
+    /// `non-disruptive-conformance`, we default to `quick` to make a quick test the most ergonomic.
+    #[structopt(long, default_value = "quick")]
+    sonobuoy_mode: Mode,
+
+    /// The kubernetes version (with or without the v prefix). Examples: v1.21, 1.21.3, v1.20.1
+    #[structopt(long)]
+    kubernetes_version: Option<K8sVersion>,
+
+    /// The kubernetes conformance image used for the sonobuoy test.
+    #[structopt(long)]
+    kubernetes_conformance_image: Option<String>,
+
+    /// The name of the secret containing aws credentials.
+    #[structopt(long)]
+    aws_secret: SecretName,
+
+    /// The name of the secret containing vsphere credentials.
+    #[structopt(long)]
+    vsphere_secret: SecretName,
+
+    /// The name of the secret containing wireguard credentials.
+    #[structopt(long)]
+    wireguard_secret: SecretName,
+
+    /// The name of the vsphere cluster that will be used (whether it is being created or already
+    /// exists).
+    #[structopt(long)]
+    cluster_name: String,
+
+    /// The ova name to use for cluster nodes.
+    #[structopt(long)]
+    ova_name: String,
+
+    /// The name of the TestSys resource that will represent EC2 instances serving as cluster nodes.
+    /// Defaults to `cluster-name-instances`.
+    #[structopt(long)]
+    vm_resource_name: Option<String>,
+
+    /// The container image of the EC2 resource provider.
+    // TODO - provide a default on ECR Public
+    #[structopt(long)]
+    vm_provider_image: String,
+
+    /// Name of the pull secret for the EC2 provider image.
+    #[structopt(long)]
+    vm_provider_pull_secret: Option<String>,
+
+    /// The number of vm nodes to launch.
+    #[structopt(long)]
+    vm_count: Option<i32>,
+
+    /// Url for tuf repo metadata.
+    #[structopt(long)]
+    tuf_repo_metadata_url: String,
+
+    /// Url for tuf repo targets.
+    #[structopt(long)]
+    tuf_repo_targets_url: String,
+
+    /// URL of the vCenter instance to connect to
+    #[structopt(long)]
+    host_url: String,
```

Comment:
Host is too generic here

```suggestion
    vcenter_url: String,
```